src/libstd/sys/common/condvar.rs

```diff
@@ -18,14 +20,14 @@ use sys::condvar as imp;
 /// condition variables. It is consequently entirely unsafe to use. It is
 /// recommended to use the safer types at the top level of this crate instead of
 /// this type.
-pub struct Condvar(imp::Condvar);
+pub struct Condvar(Box<imp::Condvar>);
```

Comment:
The initial intention of these `sys_common` types was to be a thin a layer over the system primitives as possible, unsafety an all. Could this box be left upwards and the initialization be delayed like with mutexes?